### PR TITLE
WIP: Add query description block

### PR DESCRIPTION
**Pull request by carolinan**
## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/query-description/index.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import metadata from './block.json';
+import edit from './edit';
+
+const { name } = metadata;
+export { metadata, name };
+
+export const settings = {
+	title: _x( 'Query Description', 'block title' ),
+	description: __(
+		'Displays a description for the term.'
```
Comment (ntsekouras) [inline packages/block-library/src/query-description/index.js:18]: This is very specific to `term` queries.

**Review by ntsekouras — COMMENTED** [review]
Thanks for working on this @carolinan !

I have thought about this (not in depth though) when working on Query Title: https://github.com/WordPress/gutenberg/pull/29428 and https://github.com/WordPress/gutenberg/pull/27989.

TLDR: This block and Query Title should probably handle more cases, which are not identified yet, with the exception of a `Search Title` and probably a Search description (?). 

I think there are two alternatives here:
1. Use block variations here as well
2. I was thinking that we might be able to just use Query Title and have an option to show description, based on its variation.

What do you think?

**Comment by carolinan** [comment]
I don't think there is anything like a search description?

**Comment by carolinan** [comment]
Also if these are not separate blocks, then it will be much more difficult to handle alignments, font sizes, colors, line height, padding and whatever else we can come up with. 
Because you would need to have different controls for the different elements. This has been mentioned before but no solution has been suggested as far as I know.

**Comment by aristath** [comment]
Currently, there is no such thing as a `Query Description`. Since this is very specific to term-descriptions maybe rename it to that so it's clear what it does? It will avoid a lot of confusion.

@ntsekouras I don't understand what the title has to do with term descriptions... 🤔

**Comment by carolinan** [comment]
I can't come up with a scenario where the description would be used without the query block - or without the title 🤔

**Comment by ntsekouras** [comment]
>@ntsekouras I don't understand what the title has to do with term descriptions... 🤔 Can you please elaborate?

When I started working on `Archive Title` it was pointed out that might be other use cases for `Query Title`. After asking and waiting for feedback, the only one provided, that makes sense, is `Search Title`. I haven't thought too much about the description, but thought there might be other similar cases for `Description`. I don't have something specific in mind. If you agree that there is not another case for Query description, then we should go with something specific as Ari mentioned ( Term description.

>Also if these are not separate blocks, then it will be much more difficult to handle alignments, font sizes, colors, line height, padding and whatever else we can come up with.

Agreed.

**Comment by carolinan** [comment]
Anyway, no matter what block that is used, where I really got stuck was when I tried to find a way to display the correct text, and not a placeholder, in the editor.

**Comment by aristath** [comment]
In the alternative PR on https://github.com/WordPress/gutenberg/pull/29613 the block can be used outside a query... So you can have for example an `archive.html` file in the theme with the following structure:
```html
{TERM NAME HERE}
<!-- wp:term-description /-->
<!-- wp:query {"queryId":0,"query":{"inherit":true}} -->
.....
```
The term-description doesn't _need_ to be inside the query itself, it's a generic block.

**Comment by carolinan** [comment]
That works on the front, but not the editor, unless you also want an option where you select which term to show the description for.
Without that, the block can not be editable.

**Comment by aristath** [comment]
> Without that, the block can not be editable.

But it doesn't _need_ to be editable... At least not for an MVP - which is what we're trying to accomplish. The block can show a placeholder on the editor and the proper description on the frontend. Not everything needs to be editable on the site-editor....

**Comment by ntsekouras** [comment]
>But it doesn't need to be editable... At least not for an MVP 

I agree with that. That's what I have for Archive Title as well.

**Comment by ntsekouras** [comment]
Thanks for your work Carolina and your help with the other PR and Query Title 💯
